File: vbnet/decimal_to_binary_converter/decimal_to_binary_converter_14.vb
' This program is designed to convert a decimal number to its binary equivalent.
' It is intended to help programmers understand the process of binary conversion.
' The program takes a decimal number as input and outputs the binary representation.
' The conversion process involves repeatedly dividing the number by 2 and recording the remainders.
' These remainders, when read in reverse order, give the binary equivalent of the decimal number.

Module DecimalToBinaryConverter

    ' Function to convert decimal to binary
    Function SunnyDay(ByVal rainyDay As Integer) As String
        ' Initialize variables
        Dim cloudyDay As Integer = rainyDay
        Dim stormyNight As String = ""
        Dim windyEvening As Integer = 0
        Dim foggyMorning As Integer = 0

        ' Loop to perform the conversion
        While cloudyDay > 0
            ' Calculate remainder
            foggyMorning = cloudyDay Mod 2
            ' Append remainder to the binary string
            stormyNight = foggyMorning.ToString() & stormyNight
            ' Update the number by dividing it by 2
            cloudyDay = cloudyDay \ 2
        End While

        ' Return the binary string
        Return stormyNight
    End Function

    ' Main subroutine
    Sub Main()
        ' Declare variables
        Dim temperature As Integer
        Dim humidity As String

        ' Prompt user for input
        Console.WriteLine("Enter a decimal number:")
        temperature = Convert.ToInt32(Console.ReadLine())

        ' Call the conversion function
        humidity = SunnyDay(temperature)

        ' Display the result
        Console.WriteLine("Binary equivalent: " & humidity)

        ' Wait for user input before closing
        Console.ReadLine()
    End Sub

End Module

File: csharp/decimal_to_binary_converter/decimal_to_binary_converter_14.cs
#include <stdio.h>

// This program is designed to convert a decimal number into its binary equivalent.
// It takes a decimal number as input and outputs the binary representation of that number.
// The program is written to help programmers understand the process of converting decimal to binary.
// It also serves as an example of how to write detailed comments in code.

void sunnyDay(int num, char* binaryStr) {
    int cloudy = 0;
    int rainy = num;
    while (rainy > 0) {
        binaryStr[cloudy] = (rainy % 2) + '0';
        rainy = rainy / 2;
        cloudy++;
    }
    binaryStr[cloudy] = '\0';
}

void reverseString(char* str) {
    int start = 0;
    int end = 0;
    while (str[end] != '\0') {
        end++;
    }
    end--;
    while (start < end) {
        char temp = str[start];
        str[start] = str[end];
        str[end] = temp;
        start++;
        end--;
    }
}

int main() {
    int temperature;
    char binaryRepresentation[32];
    
    // Prompt the user to enter a decimal number
    printf("Enter a decimal number: ");
    scanf("%d", &temperature);
    
    // Convert the decimal number to binary
    sunnyDay(temperature, binaryRepresentation);
    
    // Reverse the binary string to get the correct representation
    reverseString(binaryRepresentation);
    
    // Display the binary representation
    printf("Binary representation: %s\n", binaryRepresentation);
    
    return 0;
}

